test/jenkins_tests/multi_node_docker_test.py

```diff
@@ -0,0 +1,233 @@
+    stdout_data, _ = wait_for_output(proc)
+    stopped_container_id = self._get_container_id(stdout_data)
+    if not container_id == stopped_container_id:
+      raise Exception("Failed to stop container {}.".format(container_id))
+
+    proc = subprocess.Popen(["docker", "rm", "-f", container_id],
+                            stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout_data, _ = wait_for_output(proc)
+    removed_container_id = self._get_container_id(stdout_data)
+    if not container_id == stopped_container_id:
+      raise Exception("Failed to remove container {}.".format(container_id))
+
+    print("stop_node", {"container_id": container_id,
+                        "is_head": container_id == self.head_container_id})
+
+  def stop_ray(self):
+    """Stop the Ray cluster."""
+    self._stop_node(self.head_container_id)
+    for container_id in self.worker_container_ids:
+      self._stop_node(container_id)
+
+  def run_test(self, test_script, run_in_docker=False):
+    """Run a test script.
+
+    Run a test using the Ray cluster.
+
+    Args:
+      test_script: The test script to run.
+      run_in_docker: If true then the test script will be run in a docker
+        container. If false, it will be run regularly.
+
+    Returns:
+      A dictionary with information about the test script run.
+    """
+    print("Starting to run test script {}.".format(test_script))
+    proc = subprocess.Popen(["docker", "exec", self.head_container_id,
+                             "/bin/bash", "-c",
+                             "RAY_REDIS_ADDRESS={}:6379 "
+                             "python {}".format(self.head_container_ip,
+                                                test_script)],
+                            stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout_data, stderr_data = wait_for_output(proc)
+    print("STDOUT:")
+    print(stdout_data)
+    print("STDERR:")
+    print(stderr_data)
+    return {"success": proc.returncode == 0, "return_code": proc.returncode}
+
+if __name__ == "__main__":
+  parser = argparse.ArgumentParser(description="Run multinode tests in Docker.")
+  parser.add_argument("--docker-image", default="ray-project/deploy",
+                      help="docker image")
+  parser.add_argument("--mem-size", help="memory size")
+  parser.add_argument("--shm-size", default="1G", help="shared memory size")
+  parser.add_argument("--num-nodes", default=1, type=int,
+                      help="number of nodes to use in the cluster")
+  parser.add_argument("--test-script", required=True, help="test script")
+  parser.add_argument("--development-mode", action="store_true",
+                      help="use local copies of the test scripts")
+  args = parser.parse_args()
+
+  d = DockerRunner()
+  d.start_ray(mem_size=args.mem_size, shm_size=args.shm_size,
+              num_nodes=args.num_nodes, docker_image=args.docker_image,
+              development_mode=args.development_mode)
+  time.sleep(2)
```

Comment:
I just tried it out, it works like I said with the caveat that `ray.init` only retries for a few seconds and then raises an exception.